architecture/2025-06-compute-extensibility-feature-spec.md

```diff
@@ -0,0 +1,488 @@
+
+### Detailed user experience
+ <!-- <List of steps the user goes through from the start to the end of the scenario to provide more detailed view of exactly what the user is able to do given the new capabilities>  -->
+<!-- Step 1
+Step 2
+… -->
+#### Setting up and deploying an application to a Radius environment using existing Recipes for core types:
+
+1.  **Discover existing Recipes**:
+    *   Find community-provided recipes for desired platforms (e.g., ACI, AWS Fargate) from an OCI registry or Terraform module repo.
+1.  **Initialize Workspace & Environment (as Platform Engineer)**:
+    *   Use `rad workspace`, `group`, and `environment` commands to create a Radius Environment and/or Workspace. 
+    *   Define an environment in a Bicep file and then deploy it using `rad deploy env.bicep`. The new environment version (e.g., `Applications.Core/environments@2025-05-01-preview`) will not have a hard-coded `compute` kind. For example, an ACI environment might look like:
+        ```diff
+        extension radius
+
+        resource env 'Applications.Core/environments@2025-05-01-preview' = {
+            name: 'my-aci-env'
+            properties: {
+        -       compute: {
+        -           // compute kind is no longer hard-coded here
+        -       }
+                recipes: {
+                    'Applications.Core/containers': {
+                        default: {
+                        templateKind: 'bicep'
+                        plainHttp: true
+                        templatePath: 'ghcr.io/radius-project/recipes/azure/aci-container:latest'
+                        parameters: {
+                            defaultCpu: 1
+                            defaultMemoryInGB: 2
+                        }
+                        }
+                    }
+                    'Applications.Core/gateways': {
+                        default: {
+                            templateKind: 'bicep'
+                            plainHttp: true
+                            templatePath: 'ghcr.io/radius-project/recipes/azure/aci-gateway:latest'
+                            parameters: {
+                                defaultCpu: 1
+                                defaultMemoryInGB: 2
+                            }
+                        }
+                    }
```

Comment:
This example nicely demonstrates how recipes can be packaged with an environment. Do we need recipe packs if we have this capability? For example, if I want to deploy to XYZ platform, I can define an environment that contains all the recipes needed to deploy to that platform.